src/overlays/actors/ovl_Bg_Bombwall/z_bg_bombwall.c

```diff
@@ -36,29 +61,192 @@ const ActorInit Bg_Bombwall_InitVars = {
+    }
+    osSyncPrintf("(field keep 汎用爆弾壁)(arg_data 0x%04x)(angY %d)\n", this->dyna.actor.params,
+                 this->dyna.actor.shape.rot.y);
+}
+
+void BgBombwall_DestroyInternal(BgBombwall* this, GlobalContext* globalCtx) {
+    if (this->unk_2A2 & 2) {
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        this->unk_2A2 &= 0xfffd;
+    }
+    if (this->unk_2A2 & 1) {
+        Collider_DestroyTris(globalCtx, &this->collider);
+        this->unk_2A2 &= 0xfffe;
+    }
+}
+
+void BgBombwall_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgBombwall_DestroyInternal((BgBombwall*)thisx, globalCtx);
+}
+
+void func_8086EB5C(BgBombwall* this, GlobalContext* globalCtx) {
+    s16 rand;
+    s16 rand2;
+    Vec3f sp88;
+    s32 i;
+    f32 sin = Math_Sins(this->dyna.actor.shape.rot.y);
+    f32 cos = Math_Coss(this->dyna.actor.shape.rot.y);
+    Vec3f* pos = &this->dyna.actor.posRot.pos;
+    f32 temp;
+    float new_var;
+    for (i = 0; i < 7; i++) {
+        new_var = D_8086F010[i].x;
+        temp = new_var * cos;
+        sp88.x = ((sin * D_8086F010[i].z) + ((f32)temp)) + pos->x;
+        sp88.y = pos->y + D_8086F010[i].y;
+        sp88.z = ((D_8086F010[i].z * cos) - (sin * D_8086F010[i].x)) + pos->z;
+        rand = ((s16)(Math_Rand_ZeroOne() * 120.0f)) + 0x14;
+        rand2 = ((s16)(Math_Rand_ZeroOne() * 240.0f)) + 0x14;
+        func_80033480(globalCtx, &sp88, 50.0f, 2, rand, rand2, 1);
+    }
+
+    sp88.x = pos->x;
+    new_var = pos->y + 90.0f;
+    sp88.y = pos->y + 90.0f;
+    sp88.z = pos->z + 15.0f;
+    func_80033480(globalCtx, &sp88, 40.0f, 4, 0xA, 0x32, 1);
+}
+
+void BgBombwall_setFuncPointers(BgBombwall* this, GlobalContext* globalCtx) {
+    this->funcPtr = &D_05003FC0;
+    this->actionFunc = BgBombwall_actionFunc;
+}
+
+void BgBombwall_actionFunc(BgBombwall* this, GlobalContext* globalCtx) {
+    u8 temp_v0;
+
+    temp_v0 = this->collider.base.acFlags;
+    if ((temp_v0 & 2) != 0) {
+        this->collider.base.acFlags = temp_v0 & 0xFFFD;
+        this = this;
+        func_8086EDFC(this, globalCtx);
+        Flags_SetSwitch(globalCtx, this->dyna.actor.params & 0x3F);
+        return;
+    }
+    if (this->dyna.actor.xzDistFromLink < 600.0f) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, (Collider*)&this->collider);
+    }
+}
+
+void func_8086EDFC(BgBombwall* this, GlobalContext* globalCtx) {
+    this->funcPtr = &D_05003FC0;
+    this->unk_2A0 = (u16)1;
+    func_8086EB5C(this, globalCtx);
+    this->actionFunc = (void (*)(struct BgBombwall*, GlobalContext*)) & func_8086EE40;
+}
+
+void func_8086EE40(BgBombwall* this, GlobalContext* globalCtx) {
+    s16 temp_v0;
+
+    temp_v0 = this->unk_2A0;
+    if ((s32)temp_v0 > 0) {
+        this->unk_2A0 = (s16)(temp_v0 - 1);
+        return;
+    }
+    func_8086EE94(this, globalCtx);
+    if ((((s32)this->dyna.actor.params >> 0xF) & 1) != 0) {
+        func_80078884((u16)0x4802U);
+    }
+}
```

Comment:
a few things here
-this temp is probably not real
-you can do -- on the variable directly
-the early return is more likely an else
-the cast on params is probably not needed
-the arg for that last call is a sound effect id, which you can find in sfx.h

also, you can name 2A0 `timer` if you want
```suggestion
void func_8086EE40(BgBombwall* this, GlobalContext* globalCtx) {
    if (this->unk_2A0 > 0) {
        this->unk_2A0--;
    } else {
        func_8086EE94(this, globalCtx);
        if (((this->dyna.actor.params >> 0xF) & 1) != 0) {
            func_80078884(NA_SE_SY_CORRECT_CHIME);
        }
    }
}
```